crates/backend/src/encode.rs

```diff
@@ -236,23 +236,19 @@ fn shared_function<'a>(func: &'a ast::Function, _intern: &'a Interner) -> Functi
     }
 }
 
-fn shared_enum<'a>(e: &'a ast::Enum, intern: &'a Interner) -> Enum<'a> {
+fn shared_enum(e: &ast::Enum) -> Enum {
     Enum {
         name: &e.js_name,
         signed: e.signed,
-        variants: e
-            .variants
-            .iter()
-            .map(|v| shared_variant(v, intern))
-            .collect(),
+        variants: e.variants.iter().map(shared_variant).collect(),
         comments: e.comments.iter().map(|s| &**s).collect(),
         generate_typescript: e.generate_typescript,
     }
 }
 
-fn shared_variant<'a>(v: &'a ast::Variant, intern: &'a Interner) -> EnumVariant<'a> {
+fn shared_variant(v: &ast::Variant) -> EnumVariant {
     EnumVariant {
-        name: intern.intern(&v.name),
+        name: v.js_name.as_str(),
```

Comment:
Do you know what the `Interner` does exactly?

Reply:
The entire encoded AST is "borrowed" in that all strings are `&str`. (I think this is because the encoder doesn't support anything else, but I'm not sure.) The job of `intern` to turn a temporary `String` into `&'a str`. Basically, the temporary string from an identifier doesn't live long enough, so `intern` stores it somehow and returns a reference to it that does live long enough.